commands/create.go

```diff
@@ -513,31 +539,72 @@ func updateUserdataFile(driverOpts *rpcdriver.RPCFlags, machineName, userdataFla
 			return err
 		}
 	}
+	log.Infof("[updateUserdataFile] userdataFile has contents [%s]", userdataFile)
 
 	customScriptContent, err := ioutil.ReadFile(customInstallScript)
 	if err != nil {
 		return err
 	}
 	// Remove the shebang
 	customScriptContent = regexp.MustCompile(`^#!.*\n`).ReplaceAll(customScriptContent, nil)
+	log.Infof("[updateUserdataFile] customScriptContent has contents [%v]", customScriptContent)
 
 	modifiedUserdataFile, err := ioutil.TempFile("", "modified-user-data")
 	if err != nil {
-		return err
+		//return err
+		return fmt.Errorf("[updateUserdataFile] unable to created tempfile [%v]\nError returned\n[%v]", modifiedUserdataFile, err)
 	}
 	defer modifiedUserdataFile.Close()
 
-	if err := replaceUserdataFile(machineName, userdataContent, customScriptContent, modifiedUserdataFile); err != nil {
+	if err := replaceUserdataFile(machineName, osFlag, userdataContent, customScriptContent, modifiedUserdataFile); err != nil {
 		return err
 	}
+	log.Infof("[updateUserdataFile] userdata file replaced for machineName [%s]", machineName)
 
 	driverOpts.Values[userdataFlag] = modifiedUserdataFile.Name()
 
 	return nil
 }
 
-func writeCloudConfig(machineName, encodedData string, cf map[interface{}]interface{}, newUserDataFile *os.File) error {
+func writeCloudConfig(machineName, encodedData, osFlag string, cf map[interface{}]interface{}, newUserDataFile *os.File) error {
 	// Add to the write_files directive
+	// TODO: Need to check the driver GuestOS and have a switch for windows/linux
+
+	if osFlag == "windows" {
```

Comment:
maybe another switch with cases again? rather future proof it even if we only have two optinos today

Reply:
And a lot of the code in this blocked is duplicated in the linux case. We should try to consolidate this.